dante/osomerank/audience_diversity.py

```diff
@@ -136,9 +137,15 @@ def load_ad_data():
                 visitor_var_quantile.append(0)
         _DF["visitor_var"] = visitor_var_quantile
         _PAT = _list_to_pattern(*_DF.index)
+    except (configparser.NoSectionError, OSError) as e:
+        logger.error(f"Failed to load audience diversity data or prediction model. {e}")
+        raise
```

Comment:
This catcher needs to be split and separated from this `try: ... except: ...` block. The `NoSectionError` can only be triggered by L95 (`fn = config.get("AUDIENCE_DIVERSITY", "audience_diversity_file")`, so you need to wrap that line only with a try/except. I am not sure if the OSError is also something triggered by `config.get(...)` but if that's the case you can move it as well. Otherwise you'll need to find the actual function/method call that is likely to generate it, and wrap that one into a try/except.